### DISCUSSION TITLE
Update `GrantRoundManager` to inherit from Uniswap's `SwapRouter`, instead of call an external router

### DISCUSSION THREAD

**Discussion by mds1** [dcgtc/dgrants#74] [Architecture & Design] [answered]
This saves gas because:
1. We no longer need an external call
2. We no longer need to transfer tokens to/from the Router, which will reduce costs significantly

**Comment by thelostone-mc** [comment]
This is done at the moment and we have implmented a donate method which leverages SwapRouter

**Reply by mds1** [reply]
I don't think this has been done? For #76 we're calling an external router, instead of making the GrantRoundManager itself a router

**Comment by mds1** [comment, marked as answer]
This has been converted into an issue here: https://github.com/dcgtc/dgrants/issues/99
